### ISSUE TITLE
04. PR1: no-augmentation 条件の identity output contract test を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#423]
# 04. PR1: no-augmentation 条件の identity output contract test を追加する

## 目的

PR1 で追加した `file_index`, `primary_key_id`, `secondary_key_id`, `anchor_raw_indices_view` が、augmentation なしの基本条件で正しく出力されることを regression test で固定する。

## 依存 issue

- `01_issue_define_trace_identity_helpers.md`
- `02_issue_emit_file_and_key_ids.md`
- `03_issue_emit_anchor_raw_indices_view.md`

## 対象ファイル

```text
packages/seisai-engine/tests/test_fbpick_coarse.py
```

必要なら import を追加する。

### 1. 単一ファイル sample の identity field test を追加する

既存 helper `_build_global_anchor_train_sample()` を使って、augmentation なし sample を作る。

推奨 test 名:

```python
def test_global_anchor_train_dataset_adds_trace_identity_fields(
    tmp_path: Path,
) -> None:
    ...
```

検証内容:

```python
assert 'file_index' in sample
assert 'primary_key_id' in sample
assert 'secondary_key_id' in sample
assert 'anchor_raw_indices_view' in sample

assert sample['file_index'].dtype == torch.int64
assert sample['primary_key_id'].dtype == torch.int64
assert sample['secondary_key_id'].dtype == torch.int64
assert sample['anchor_raw_indices_view'].dtype == torch.int64

assert tuple(sample['anchor_raw_indices_view'].shape) == (256,)
```

mapping は output の文字列 field から検証する。

```python
primary_map = {'ffid': 0, 'chno': 1, 'cmp': 2}
secondary_map = {'ffid': 0, 'chno': 1, 'offset': 2}

assert int(sample['primary_key_id']) == primary_map[sample['key_name']]
assert int(sample['secondary_key_id']) == secondary_map[sample['secondary_key']]
```

no augmentation では view row と pre-transform row が一致するため、`anchor_raw_indices_view` は `anchor_raw_indices` と一致する。

```python
np.testing.assert_array_equal(
    sample['anchor_raw_indices_view'].numpy(),
    sample['anchor_raw_indices'].numpy(),
)
```

さらに invalid row の `-1` を明示的に確認する。

```python
trace_valid = np.asarray(sample['meta']['trace_valid'], dtype=np.bool_)
assert np.all(sample['anchor_raw_indices_view'].numpy()[~trace_valid] == -1)
```

### 2. 複数ファイルで `file_index` が 0-based になることを確認する

`build_labeled_infer_dataset(... anchor_mode='center' ...)` は `__getitem__(index)` で `index % len(file_infos)` の file を選ぶため、複数ファイルの `file_index` を deterministic に確認しやすい。

推奨 test 名:

```python
def test_global_anchor_labeled_dataset_file_index_matches_file_infos_order(
    tmp_path: Path,
) -> None:
    ...
```

実装方針:

- 2 つの小さな SEG-Y と fb npy を作る。
- `build_labeled_infer_dataset()` で dataset を作る。
- `ds[0]['file_index'] == 0` を確認する。
- `ds[1]['file_index'] == 1` を確認する。
- `ds.close()` は `finally` で呼ぶ。

既存 test の書き方に合わせる。

## 非目標

この issue では以下を test しない。

- `hflip` 時の row alignment
- `space.factor_h` 時の row alignment
- memory consistency loss
- train loop integration

hflip / factor_h は issue 05 で扱う。

## 受け入れ条件

- no augmentation の train sample で、追加 identity field がすべて存在する。
- 追加 field の dtype/shape が固定される。
- `primary_key_id` / `secondary_key_id` が既存文字列 output と一致する。
- `anchor_raw_indices_view` が no augmentation で `anchor_raw_indices` と一致する。
- invalid row が `-1` で mask されている。
- 複数ファイル validation/labeled dataset で `file_index` が `self.file_infos` の順序と一致する。

## 確認コマンド

追加 test を含めて以下を通す。

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py -q
```

時間が厳しい場合は、追加した test 名を `-k` で絞って先に確認する。

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py \
  -k 'trace_identity_fields or file_index_matches_file_infos_order' \
  -q
```
